### PR TITLE
Adjust HS tendency to account for pressure with surface elevation

### PR DESCRIPTION
**Pull request by akshaysridhar**
# PULL REQUEST

## Purpose and Content
Update tendency so pressure variable accounts for surface elevation

## Benefits and Risks
Benefit: Generalises to surfaces with warp / topography
Risks: No foreseeable risks - this should be a non-breaking change for existing tests. 

## Linked Issues
#408 

## PR Checklist
- [x] This PR has a corresponding issue OR is linked to an SDI.
- [x] I have followed CliMA's codebase [contribution](https://clima.github.io/ClimateMachine.jl/latest/Contributing/) and [style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines OR N/A.
- [x] I have followed CliMA's [documentation policy](https://github.com/CliMA/policies/wiki/Documentation-Policy).
- [x] I have checked all issues and PRs and I certify that this PR does not duplicate an open PR.
- [x] I linted my code on my local machine prior to submission OR N/A.
- [x] Unit tests are included OR N/A. (Integration buildkite tests)
- [x] Code used in an integration test OR N/A.
- [x] All tests ran successfully on my local machine OR N/A.
- [x] All classes, modules, and function contain docstrings OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: examples/hybrid/sphere/baroclinic_wave_utilities.jl [hunk 1]
```diff
@@ -408,7 +413,10 @@ function held_suarez_tendency!(Yₜ, Y, p, t)
     Δθ_z = FT(10)
     T_min = FT(200)
 
-    @. ᶜσ = ᶜp / MSLP
+    p_int_size = size(parent(ᶜp))
+    p_sfc = reshape(parent(p_sfc), (1, p_int_size[2:end]...))
+    parent(ᶜσ) .= parent(ᶜp) ./ parent(p_sfc)
```
Comment (szy21) [inline examples/hybrid/sphere/baroclinic_wave_utilities.jl:418]: do we need parent here?

**Review by szy21 — APPROVED** [review]

File: examples/hybrid/sphere/baroclinic_wave_utilities.jl (hunk 1 above)
Comment (akshaysridhar) [reply examples/hybrid/sphere/baroclinic_wave_utilities.jl:418]: Yeah I ran into the inconsistent `space instance` error otherwise

**Comment by szy21** [comment]
Please squash the commits and merge:)

**Comment by akshaysridhar** [comment]
bors try

**Pull request merged**